tests/common.sh

```diff
@@ -103,16 +103,24 @@ function install_ide()
 {
     local ide_path=$1
     local core_path=$2
+    local debug=$3
     wget -O arduino.tar.xz https://www.arduino.cc/download.php?f=/arduino-nightly-linux64.tar.xz
     tar xf arduino.tar.xz
     mv arduino-nightly $ide_path
     cd $ide_path/hardware
     mkdir esp8266com
     cd esp8266com
     ln -s $core_path esp8266
+    local debug_flags=""
+    if [ "$debug" = "debug" ]; then
+        debug_flags="-DDEBUG_ESP_PORT=Serial -DDEBUG_ESP_SSL -DDEBUG_ESP_TLS_MEM -DDEBUG_ESP_HTTP_CLIENT -DDEBUG_ESP_HTTP_SERVER -DDEBUG_ESP_CORE -DDEBUG_ESP_WIFI -DDEBUG_ESP_HTTP_UPDATE -DDEBUG_ESP_UPDATER -DDEBUG_ESP_OTA -DDEBUG_ESP_OOM"
+    fi
     # Set custom warnings for all builds (i.e. could add -Wextra at some point)
-    echo "compiler.c.extra_flags=-Wall -Werror" > esp8266/platform.local.txt
-    echo "compiler.cpp.extra_flags=-Wall -Werror" >> esp8266/platform.local.txt
+    echo "compiler.c.extra_flags=-Wall -Werror $debug_flags" > esp8266/platform.local.txt
+    echo "compiler.cpp.extra_flags=-Wall -Werror $debug_flags" >> esp8266/platform.local.txt
```

Comment:
should `-Wextra`have its place here too ?

Reply:
Sure.  It wasn't in the original "Add all warnings" patch but we can pull it in here...